### PR TITLE
Issue #140 - Fixed parsing of parameter values that contain commas that have been properly escaped using a backslash.

### LINKED ISSUE #140: List type parameters, which contain commas, are not processed correctly.
Apparently, `cfn-lint` cannot handle parameters with types `CommaDelimitedList` and `List<Number>`.

Steps to reproduce:
1) Given the following template:
```
Parameters:
  SomeParam:
    Type: CommaDelimitedList
Resources:
  bucket_name:
    Type: AWS::S3::Bucket
    Properties:
      BucketName:
        Fn::Select:
          - 1
          - Ref: SomeParam
      Tags:
        Ref: SomeParam
```

2) We can observe that `cfn-lint` does not correctly parse the provided parameter value:
```
$ cfn-lint validate testData/valid/yaml/parameters_type_commadelimitedlist.yaml -p SomeParam="test,dev,prod"
0 infos
0 warn
2 crit
Resource: Resources > bucket_name > Properties > BucketName
Message: First element of Fn::Select exceeds the length of the list.
Documentation: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-select.html

Resource: Resources > bucket_name > Properties > Tags > 0
Message: Expecting an object, got 'test'
Documentation: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket.html

Template invalid!
```

### PR DESCRIPTION
**Pull request by RazzM13**
#140 - Fixed parsing of parameter values that contain commas that have been properly escaped using a backslash.

### CODE REVIEW

**Comment by RazzM13** [comment]
This is not ready yet, tests are coming but that's going to have to wait until tomorrow.

**Comment by RazzM13** [comment]
This should be ready for merging.

File: src/index.ts [hunk 1]
```diff
@@ -11,7 +11,17 @@ let firstArg: string | undefined = undefined
 let secondArg: string = undefined!;
 
 function list(val: string) {
```
Comment (martysweet) [inline src/index.ts]: Comments explaining the use of this please, where is this used?

File: src/test/indexTest.ts [hunk 2]
```diff
@@ -118,6 +118,24 @@ describe('index', () => {
                 done();
             });
         }).timeout(5000);
+
+        it('parameters of the CommaDelimitedList type should accept lists as values', (done) => {
+            exec('node lib/index.js validate testData/valid/yaml/parameters_type_commadelimitedlist.yaml --parameters SomeParam="test\\,dev\\,prod"', function(error, stdout, stderr) {
```
Comment (martysweet) [inline src/test/indexTest.ts]: Can this be mentioned in the docs?

something\,something\,something is possible

**Comment by RazzM13** [comment]
Hi @martysweet, I'm guessing this should be complete... anything else? :smiley:

**Comment by martysweet** [comment]
Looks great!

**Comment by akdor1154** [comment]
@RazzM13 were the additions to aws_parameter_types.json necessary? I'm not sure they were needed (at least they shouldn't have been, it's quite possible my code was dodgy and doesn't work though :) )

**Comment by martysweet** [comment]
@akdor1154 Can you create an issue for this and reference this PR? Otherwise, it will get lost. :grinning:

**Comment by RazzM13** [comment]
Hi @akdor1154, I believe that I refactored that particular part and it does appear connected now:

https://github.com/RazzM13/cfn-lint/blob/b17076f355113570048637c29d31b9e0e0b759c5/src/awsData.ts#L100

https://github.com/RazzM13/cfn-lint/blob/b17076f355113570048637c29d31b9e0e0b759c5/src/validator.ts#L7-L11

https://github.com/RazzM13/cfn-lint/blob/b17076f355113570048637c29d31b9e0e0b759c5/src/validator.ts#L212

**Pull request merged**